src/modules/Builder/features/ActivityItemsFlow/ItemFlow/ItemFlow.tsx

```diff
@@ -17,15 +17,12 @@ export const ItemFlow = ({ name, index, onRemove }: ItemFlowProps) => {
   const conditionsName = `${itemName}.conditions`;
   const dataTestid = `builder-activity-item-flow-${index}`;
 
-  const { control, getFieldState } = useFormContext();
-  const {
-    fields: conditions,
-    append: appendCondition,
-    remove: removeCondition,
-  } = useFieldArray({
+  const { control, getFieldState, watch } = useFormContext();
+  const { append: appendCondition, remove: removeCondition } = useFieldArray({
     control,
     name: conditionsName,
   });
+  const conditions = watch(conditionsName);
```

Comment:
Do we really need it there just for passing it through props? Maybe it does make sense to move it to ItemFlowContent?

Reply:
good point, I will check